### PR TITLE
Simple filtering 

### PR DESCRIPTION
**Pull request by SamuelJanas**
Currently the site looks the following way:
image
The sidebar can be hidden for better viewing experience. 
## Filtering:
- **Pieces per page**: change the number of pieces shown on a single page
- **Split**: uses the selected split from the dataset
- **Composer**: if the corresponding radio button is selected, shows all the pieces from the chosen composer
- **Title**: chooses only the titles that contain the value inserted in the textbox, could be changed to ignore the case but wasn't sure.
## Possible improvements:
I will be looking into a way to download the audio file on demand to make the experience smoother.

### CODE REVIEW

**Review by roszcz — COMMENTED** [review]
Thanks, this works great!

I'm thinking about turning this dashboard into a website (e.g. maestro.pianofor.ai) - and for that I think we could try to make it more responsive. Here are some ideas on how to do it:

1. Pre-generate all audio and pianorolls. This will take some time, but after that there will be no waiting for the mp3 to render.
2. Don't use HF datasets to manage data on the dashboard - I think there's some waiting time for every filtering and load operation because of how those datasets are set up internally. I think we could use `pd.DataFrame` instead - assuming that after 1. we get a data frame with all paths necessary, we should be able to do filtering like that:

```python
ids = df.title.str.contains(search_by_title, case=False)
selected_df = df[ids]
```

Let me know if this makes sense!

File: dashboard.py [hunk 1]
```diff
@@ -4,22 +4,113 @@
 
 import utils as U
 
+# Define the number of records to display per page
+n_records_per_page = 4
```
[the remark this answers is not in the document]
Comment (SamuelJanas) [reply dashboard.py:8]: There weren't any performance issues that I know of, the only one being audio processing but that would be mitigated following your suggestion to preprocess it all. The number here was strictly for cosmetic purposes. 😅

File: dashboard.py [hunk 2]
```diff
@@ -4,22 +4,113 @@
...
-cols = st.columns(2)
+def load_data(selected_split):
+    # Load the dataset for the selected split
+    dataset = load_dataset("roszcz/maestro-v1-sustain", split=selected_split)
 
+    # Apply the filtering based on the current filter option
+    if filter_option == "composer" and selected_composer:
+        dataset = dataset.filter(lambda example: example["composer"] == selected_composer)
+    elif filter_option == "title" and search_by_title:
+        dataset = dataset.filter(lambda example: str(search_by_title) in example["title"])
```
[the remark this answers is not in the document]
Comment (SamuelJanas) [reply dashboard.py:24]: That's what I was thinking, thanks for the feedback!

**Comment by SamuelJanas** [comment]
Great suggestions, I'll be implementing that and looking at the performance.

**Review by wmilkowska — APPROVED** [review]

**Pull request closed without merging**